File: JAVA/CharSort/Facade/CSFacade.java
package CharSort.Facade;

import CharSort.Entity.*;
import CharSort.Interface.*;
import CharSort.model.*;
import CharSort.*;
import java.util.*;

public final class CSFacade
{
	private static ICSModel csModel;
	public static List<InChar> CreateInCharObjects(String strInput)
	{
		IInCharObjCreator icobc = new InCharObjCreator();
		return icobc.CreateInCharObjects(strInput);
	}

	public static ICSModel GetCSModel()
	{
		if (csModel == null)
		{
			csModel = new CSModel();
		}
		return csModel;
	}

	public static List<InChar> CalculateInCharOutObjects(List<InChar> inChars)
	{
		IInCharOutObjCalculator icobc = new InCharOutObjCalculator();
		return icobc.CalculateInCharOutObjects(inChars);
	}

	public static String GetOutCharString(List<InChar> outChars)
	{
		IOutCharStringGetter icobc = new OutCharStringGetter();
		return icobc.GetOutCharString(outChars);
	}
}

File: CharSort/Facade/CSFacade.cs
using CharSort.Entity;
using CharSort.Interface;
using CharSort.model;
using System;
using System.Collections.Generic;

namespace CharSort.Facade
{
    public static class CSFacade
    {
        static ICSModel csModel;
        internal static IList<InChar> CreateInCharObjects(string strInput)
        {
            IInCharObjCreator icobc = new InCharObjCreator();
            return icobc.CreateInCharObjects(strInput);
        }

        internal static ICSModel GetCSModel()
        {
            if (csModel == null) { csModel = new CSModel(); }
            return csModel;
        }

        internal static IList<InChar> CalculateInCharOutObjects(IList<InChar> inChars)
        {
            IInCharOutObjCalculator icobc = new InCharOutObjCalculator();
            return icobc.CalculateInCharOutObjects(inChars);
        }

        internal static string GetOutCharString(IList<InChar> outChars)
        {
            IOutCharStringGetter icobc = new OutCharStringGetter();
            return icobc.GetOutCharString(outChars);
        }
    }
}
